### PR TITLE
Add checkpoint eraser

### PR DESCRIPTION
**Pull request by jodusan**
## What do these changes do?

Add checkpoint eraser which keeps only last N checkpoints. It also keeps only the last K highest-mean-reward checkpoints.

### CODE REVIEW

**Review by nikola-j — APPROVED** [review]

File: python/ray/rllib/agents/agent.py [hunk 1]
```diff
@@ -137,6 +137,10 @@
     # Similar to remote_worker_envs, but runs the envs asynchronously in the
     # background for greater efficiency. Conflicts with remote_worker_envs.
     "async_remote_worker_envs": False,
+    # Number of last seen checkpoints to keep on disk
+    "keep_checkpoints": 30,
```
Comment (bjg2) [inline python/ray/rllib/agents/agent.py:141]: move these to tune, where `checkpoint_freq` is. Follow the path that argument is going. Your arguments should be able to be used from CLI as well.

File: python/ray/rllib/agents/agent.py [hunk 2]
```diff
@@ -137,6 +137,10 @@
...
+    # Number of last seen highest rewarded checkpoints to keep on disk
+    "keep_best_checkpoints": 30,
```
Comment (bjg2) [inline python/ray/rllib/agents/agent.py:143]: Both your arguments should not change current behavior - default them to None

File: python/ray/tune/trainable.py [hunk 3]
```diff
@@ -405,6 +418,53 @@ def _restore(self, checkpoint):
 
         raise NotImplementedError
 
+    def _delete(self, checkpoint_dir):
```
Comment (bjg2) [inline python/ray/tune/trainable.py:421]: Rename _delete to delete_checkpoint, as it will be used from tune

File: python/ray/tune/trainable.py [hunk 4]
```diff
@@ -405,6 +418,53 @@ def _restore(self, checkpoint):
...
+
+        Parameters
+        ----------
+            checkpoint_dir (str): The directory where the checkpoint
+                file is placed.
+
+        """
+        shutil.rmtree(checkpoint_dir)
+
+    def _checkpoint_high_reward(self, checkpoint_dir):
```
Comment (bjg2) [inline python/ray/tune/trainable.py:432]: Move everything except delete_checkpoint to tune, probably somewhere in `ray_trial_executor.py`

**Review by bjg2 — CHANGES_REQUESTED** [review]

**Pull request closed without merging**